src/arch/xtensa/include/arch/atomic.h

```diff
@@ -45,11 +46,12 @@ static inline int32_t arch_atomic_add(atomic_t *a, int32_t value)
 		: "a" (&a->value), "a" (value)
 		: "memory");
 
-	return (*(volatile int32_t *)&a->value);
+	return ret;
```

Comment:
This should return "current" i.e. the value before the sucessful increment as the "ret" value may be stale if other threads are also incrementing the same atomic value.

Reply:
updated